src/clientHandler.ts

```diff
@@ -12,298 +10,177 @@ import {
-    let name = `Terraform LS`;
-    let wsFolder: vscode.WorkspaceFolder;
-    let rootModulePaths: string[];
-    let terraformExecPath: string;
-    let terraformExecTimeout: string;
-    let terraformLogFilePath: string;
-    let excludeModulePaths: string[];
-    let enableReferenceCountCodeLens: boolean;
-    let ignoreDirectoryNames: string[];
-    let documentSelector: DocumentSelector;
-    let outputChannel: vscode.OutputChannel;
-    if (location) {
-      channelName = `${binaryName}: ${location}`;
-      id = `terraform-ls/${location}`;
-      name = `Terraform LS: ${location}`;
-      wsFolder = getWorkspaceFolder(location);
-      documentSelector = [
-        { scheme: 'file', language: 'terraform', pattern: `${wsFolder.uri.fsPath}/**/*` },
-        { scheme: 'file', language: 'terraform-vars', pattern: `${wsFolder.uri.fsPath}/**/*` },
-      ];
-      terraformExecPath = config('terraform-ls', wsFolder).get('terraformExecPath');
-      terraformExecTimeout = config('terraform-ls', wsFolder).get('terraformExecTimeout');
-      terraformLogFilePath = config('terraform-ls', wsFolder).get('terraformLogFilePath');
-      rootModulePaths = config('terraform-ls', wsFolder).get('rootModules');
-      excludeModulePaths = config('terraform-ls', wsFolder).get('excludeRootModules');
-      enableReferenceCountCodeLens = config('terraform', wsFolder).get('enableReferenceCountCodeLens');
-      ignoreDirectoryNames = config('terraform-ls', wsFolder).get('ignoreDirectoryNames');
-      outputChannel = vscode.window.createOutputChannel(channelName);
-      outputChannel.appendLine(`Launching language server: ${cmd} ${serverArgs.join(' ')} for folder: ${location}`);
-    } else {
-      documentSelector = [
-        { scheme: 'file', language: 'terraform' },
-        { scheme: 'file', language: 'terraform-vars' },
-      ];
-      terraformExecPath = config('terraform-ls').get('terraformExecPath');
-      terraformExecTimeout = config('terraform-ls').get('terraformExecTimeout');
-      terraformLogFilePath = config('terraform-ls').get('terraformLogFilePath');
-      rootModulePaths = config('terraform-ls').get('rootModules');
-      excludeModulePaths = config('terraform-ls').get('excludeRootModules');
-      enableReferenceCountCodeLens = config('terraform').get('enableReferenceCountCodeLens');
-      ignoreDirectoryNames = config('terraform-ls').get('ignoreDirectoryNames');
-      outputChannel = vscode.window.createOutputChannel(channelName);
-      outputChannel.appendLine(`Launching language server: ${cmd} ${serverArgs.join(' ')}`);
-    }
+    const initializationOptions = this.getInitializationOptions(commandPrefix);
 
-    if (rootModulePaths.length > 0 && excludeModulePaths.length > 0) {
-      throw new Error(
-        'Only one of rootModules and excludeRootModules can be set at the same time, please remove the conflicting config and reload',
-      );
-    }
+    const serverOptions: ServerOptions = this.getServerOptions();
```

Comment:
We could get rid of the type here and add it as return type of `this.getServerOptions()`
```suggestion
    const serverOptions = this.getServerOptions();
```